dof/src/fmt.rs

```diff
@@ -92,39 +92,62 @@ fn fmt_dof_sec_type<T: Debug + FromBytes + Copy>(data: &[u8]) -> String {
         .join("\n")
 }
 
+#[derive(Clone, Copy)]
+pub enum FormatMode {
+    // Output is formatted as the Rust types used to represent DOF data throughout the `usdt` crate.
+    Pretty,
+    // The same out as `Pretty`, but formatted as JSON.
+    Json,
+    /// Underlying DOF C structs are formatted
+    Raw {
+        /// If true, the DOF section data is included, along with the secion headers.
+        /// If false, only the section headers are printed.
+        include_sections: bool,
+    },
+}
+
 /// Format all DOF data in an object file into a pretty-printable string.
 ///
-/// If `raw` is true, then the raw, underlying DOF C structs are formatted. If false, the data is
-/// formatted as the Rust types used to represent DOF data throughout the `usdt` crate.
-///
-/// If `include_sections` is true, the DOF binary section data is included, along with the section
-/// headers. If false, only the section headers are printed.
+/// Uses the `FormatMode` to determine how the data is formatted.
 ///
 /// If the file is not of the correct format, or has invalid DOF data, an `Err` is returned. If the
 /// file has no DOF data, `None` is returned.
 pub fn fmt_dof<P: AsRef<Path>>(
     file: P,
-    raw: bool,
-    include_sections: bool,
+    format: FormatMode,
```

Comment:
As stated in the description, I could've just added another bool flag here but these arguments have semantic relation and switching to an enum seemed the more practical long term solution. In the previous case `raw: false, include_sections: true` had no impact on the output which was no explicit from function's API. 

This _is_ a breaking change for the `dof` crate, but I feel it's a worth while one.

Reply:
That all seems fine to me, I have no issues changing this interface. I do think we should bump the version of the crate though, when we publish. I think it's on `0.1.x`, so we should go to `0.2.0`, probably?